solidity.pegjs

```diff
@@ -1298,14 +1298,14 @@ ModifierDeclaration
 
 FunctionDeclaration
   = FunctionToken __ fnname:FunctionName __ args:ModifierArgumentList? __
-    body:("{" __ FunctionBody __ "}")? (__ EOS)?  // Remove EmptyStatement if no body
+    body:(__ FunctionBody __)? (__ EOS)?  // Remove EmptyStatement if no body

```

Comment:
These __ around FunctionBody look redundant. Can you confirm and remove them?

Another thing we can do (feel free to put it in this pull request or a new one) is to change the optional EOS at the end with a proper choice operator: functions can have a body or an EOS at the end, not both.